src/sentry/relay/config/metric_extraction.py

```diff
@@ -1,26 +1,122 @@
 from dataclasses import dataclass
-from typing import Any, List, Literal, Sequence, Tuple, TypedDict, Union, cast
+from typing import Any, List, Literal, Optional, Sequence, Tuple, TypedDict, Union, cast
 
+from typing_extensions import NotRequired
+
+from sentry import features
 from sentry.api.endpoints.project_transaction_threshold import DEFAULT_THRESHOLD
+from sentry.constants import DataCategory
+from sentry.incidents.models import AlertRule, AlertRuleStatus
 from sentry.models import (
     Project,
     ProjectTransactionThreshold,
     ProjectTransactionThresholdOverride,
     TransactionMetric,
 )
 
+# GENERIC METRIC EXTRACTION
+
+
+# Name component of MRIs used for custom alert metrics.
+# TODO: Move to a common module.
+CUSTOM_ALERT_METRIC_NAME = "transactions/alert"
+
+# Version of the metric extraction config.
+_METRIC_EXTRACTION_VERSION = 1
+
 
-class RuleConditionInner(TypedDict):
-    op: Literal["eq", "gt", "gte"]
+RuleCondition = Union["LogicalRuleCondition", "ComparingRuleCondition", "NotRuleCondition"]
+
+
+class ComparingRuleCondition(TypedDict):
+    op: Literal["eq", "gt", "gte", "lt", "lte", "glob"]
     name: str
     value: Any
 
 
-# mypy does not support recursive types. type definition is a very small subset
-# of the values relay actually accepts
-class RuleCondition(TypedDict):
-    op: Literal["and"]
-    inner: Sequence[RuleConditionInner]
+class LogicalRuleCondition(TypedDict):
+    op: Literal["and", "or"]
+    inner: Sequence[RuleCondition]
+
+
+class NotRuleCondition(TypedDict):
+    op: Literal["not"]
+    inner: RuleCondition
+
+
+class TagSpec(TypedDict):
+    key: str
+    field: NotRequired[str]
+    value: NotRequired[str]
+    condition: NotRequired[RuleCondition]
+
+
+class MetricSpec(TypedDict):
+    category: Literal["transaction"]
+    mri: str
+    field: NotRequired[Optional[str]]
+    condition: NotRequired[RuleCondition]
+    tags: NotRequired[Sequence[TagSpec]]
+
+
+class MetricExtractionConfig(TypedDict):
+    version: int
+    metrics: List[MetricSpec]
```

Comment:
Are we postponing the addition of `tags` here?

Reply:
Keeping it simple at this point since it is neither implemented in relay (just the schema). This can be added once the functionality is used.